HW04.Rmd

```diff
@@ -77,45 +175,217 @@ Assume that:
+    geom_signif(comparisons = list(c("Not involved", "Involved")), map_signif_level = TRUE, test = wilcox.test) +
+    scale_y_continuous(breaks = seq(0, 120, 20)) +
+    labs(title = "Police involvement in gun deaths according to victim's age",
+         x = "Police involvement",
+         y = "Victim's age",
+         fill = "Police involvement",
+         caption = "Source: fivethirtyeight.com") 
+
+#A summary table.
+kable(dp_age %>%
+          group_by(police) %>%
+          summarize(median_v = median(age), min_v = min(age)), format = "markdown", col.names = c("Police involvement", "Median victim's age", "Youngest victim"))
 
 ```
 
+**Answer**
+The distribution of victim's age is not exactly normal (see the Kolmogorov-Smirnov test results, p < 0.001), but the data has homoscedasticity (see the Fligner-Killen test, p > 0.05), therefore, the Mann-Whitman-Wilcoxon test is a better option to assess statistical differences between samples than the t-test. According to the analysis, the victims' age is statistically higher when police is involved. Check the table for the median values. It is also noticeable than police officers didn't shoot anyone younger than 12.
+
 ```{r stats_2?}
 
+#Wilcoxon test for every race, age and police involvement.
+wilcox.test(deaths_police$age[deaths_police$police == "Not involved" & deaths_police$race == "AP"],
+       deaths_police$age[deaths_police$police == "Involved" & deaths_police$race == "AP"])
+
+wilcox.test(deaths_police$age[deaths_police$police == "Not involved" & deaths_police$race == "BL"],
+       deaths_police$age[deaths_police$police == "Involved" & deaths_police$race == "BL"])
+
+wilcox.test(deaths_police$age[deaths_police$police == "Not involved" & deaths_police$race == "His"],
+       deaths_police$age[deaths_police$police == "Involved" & deaths_police$race == "His"])
+
+wilcox.test(deaths_police$age[deaths_police$police == "Not involved" & deaths_police$race == "NNA"],
+       deaths_police$age[deaths_police$police == "Involved" & deaths_police$race == "NNA"])
+
+wilcox.test(deaths_police$age[deaths_police$police == "Not involved" & deaths_police$race == "WH"],
+       deaths_police$age[deaths_police$police == "Involved" & deaths_police$race == "WH"])
 
 ```
 
 
 Finally, all together now:
-```{r police_3}
+
+```{r police_3, echo =F}
+
+deaths_police %>%
+    ggplot(aes(x = race, y = age, fill = police)) +
+    geom_boxplot(alpha = 0.5, outlier.shape = NA) +
+    geom_signif(y_position = c(82, 67, 67, 67, 92), xmin = c(0.75, 1.75, 2.75, 3.75, 4.75), 
+                xmax = c(1.25,2.25,3.25, 4.25, 5.25), annotation = c("NS", "***", "***", "NS", "NS"), 
+                tip_length = 0.01) +
+    scale_y_continuous(breaks = seq(0, 120, 20)) +
+    labs(title = "Police involvement in gun deaths according to victim's age and race",
+         x = "Race",
+         y = "Victim's age",
+         fill = "Police involvement",
+         caption = "Source: fivethirtyeight.com") 

```

Comment:
Big fan of this graph, just well done especially with the plotting of the significance on the graph